exercises/practice/secret-handshake/.approaches/iterate-once/content.md

```diff
@@ -0,0 +1,97 @@
+# Iterate once
+
+```rust
+const SIGNS: [&'static str; 4] = ["wink", "double blink", "close your eyes", "jump"];
+const REVERSE_SIGNS: u8 = 16;
+
+pub fn actions(n: u8) -> Vec<&'static str> {
+    let (mut action, action_incr, end) = match n & REVERSE_SIGNS {
+        0 => (0, 1, 4),
+        _ => (3, -1, -1),
+    };
+    let mut output: Vec<&'static str> = Vec::new();
+    
+    loop {
+        if action == end {
+            break;
+        }
+        if (n & (1 << action)) != 0 {
+            output.push(SIGNS[action as usize])
+        }
+        action += action_incr
+    }
+    output
+}
```

Comment:
I must admit that I'm not well-versed in Rust, but to me this is very hard to read. Is this really an idiomatic way to solve the exercise? What about something simple like: https://exercism.org/tracks/rust/exercises/secret-handshake/solutions/aciba90?

Reply:
There are simpler approaches which don't offer as much in pedagogic value. Since I only did one approach, I used one which is not exactly unidiomatic, but which afforded more in the way of explanation.

Reply:
Right. Okay!

Reply:
It _is_ an interesting approach